pkg/proxy/util/utils.go

```diff
@@ -221,29 +221,23 @@ func GetNodeAddresses(cidrs []string, nw NetworkInterfacer) (sets.String, error)
 		}
 
 		_, ipNet, _ := net.ParseCIDR(cidr)
-		for _, itf := range itfs {
-			addrs, err := nw.Addrs(&itf)
-			if err != nil {
-				return nil, fmt.Errorf("error getting address from interface %s, error: %v", itf.Name, err)
+		for _, addr := range addrs {
+			var ip net.IP
+			switch v := addr.(type) {
```

Comment:
@aojea 
kube-proxy only run on linux, in linux, net.InterfaceAddrs will only return net.IPNet's, Do we only need to determine whether the type of addr is net.IPNet?

In addition, net.InterfaceAddrs may return net.IPAddr's or net.IPNet's.

https://github.com/golang/go/blob/4bb0847b088eb3eb6122a18a87e1ca7756281dcc/src/net/interface_linux.go#L175~L200

https://github.com/golang/go/blob/4bb0847b088eb3eb6122a18a87e1ca7756281dcc/src/net/interface_windows.go#L113~L126

Reply:
Can you comment this fact here, so nobody "fixes" it later?

Reply:
we do have a windows kubeproxy, it doesn't use this function though but who knows :shrug: 
Since we don't have platform specific code I think that we just should comment it for future reference as Tim says